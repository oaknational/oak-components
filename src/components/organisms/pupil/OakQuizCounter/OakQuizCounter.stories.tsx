import React from "react";
import { Meta, StoryObj } from "@storybook/react";

import { OakQuizCounter } from "./OakQuizCounter";

import { OakFlex } from "@/components/atoms";

const meta: Meta<typeof OakQuizCounter> = {
  component: OakQuizCounter,
  tags: ["autodocs"],
  title: "components/organisms/pupil/OakQuizCounter",
  argTypes: {
    counter: {
      control: {
        type: "number",
      },
    },
  },
  args: {
    counter: 6,
    total: 6,
  },
  decorators: [
    (Story) => (
      <OakFlex $flexDirection="column" $gap="space-between-m">
        <Story />
      </OakFlex>
    ),
  ],
};
export default meta;

type Story = StoryObj<typeof OakQuizCounter>;

export const Default: Story = {
  render: (args) => <OakQuizCounter {...args} />,
};
