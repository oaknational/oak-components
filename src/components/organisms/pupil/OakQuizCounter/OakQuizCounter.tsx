import React from "react";

import { OakSpan } from "@/components/atoms";

export type OakQuizCounterProps = {
  counter: number;
  total: number;
};
export const OakQuizCounter = (props: OakQuizCounterProps) => {
  const { counter, total } = props;
  return (
    <OakSpan $font={"heading-light-4"} $color={"text-disabled"}>
      <OakSpan $font={"heading-4"} $color={"text-primary"}>
        {counter}{" "}
      </OakSpan>
      of {total}
    </OakSpan>
  );
};
