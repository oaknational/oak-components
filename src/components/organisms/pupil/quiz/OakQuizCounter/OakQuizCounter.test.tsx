import React from "react";
import "@testing-library/jest-dom";

import { OakQuizCounter } from "./OakQuizCounter";

import renderWithTheme from "@/test-helpers/renderWithTheme";

describe(OakQuizCounter, () => {
  it("matches snapshot", () => {
    const { container } = renderWithTheme(
      <OakQuizCounter counter={5} total={6} />,
    );

    expect(container).toMatchSnapshot();
  });
});
