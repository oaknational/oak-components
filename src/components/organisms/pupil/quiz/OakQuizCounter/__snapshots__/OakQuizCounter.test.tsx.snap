// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OakQuizCounter matches snapshot 1`] = `
.c0 {
  font-family: --var(google-font),Lexend,sans-serif;
}

.c1 {
  display: flex;
  flex-direction: column;
  align-items: end;
  gap: 0.75rem;
}

.c4 {
  display: flex;
  flex-wrap: wrap;
  gap: 0.5rem;
}

.c2 {
  color: #575757;
  font-family: --var(google-font),Lexend,sans-serif;
  font-weight: 400;
  font-size: 2rem;
  line-height: 2.5rem;
  letter-spacing: 0.0115rem;
}

.c3 {
  color: #222222;
  font-family: --var(google-font),Lexend,sans-serif;
  font-weight: 600;
  font-size: 2rem;
  line-height: 2.5rem;
  letter-spacing: 0.0115rem;
}

.c5 {
  fill: #222222;
}

.c6 {
  fill: #b0e2de;
}

<div
  className="c0 c1"
>
  <span
    className="c2"
  >
    <span
      className="c3"
    >
      5
       
    </span>
    of 
    6
  </span>
  <div
    className="c0 c4"
  >
    <svg
      className="c5"
      height="7"
      width="26"
    >
      <path
        d="M25.9946 5.39444C25.6239 6.06393 25.2305 6.19303 24.8209 6.19244C22.1095 6.19103 19.4038 6.23513 16.6954 6.17886C13.8755 6.12554 11.0568 5.95577 8.23701 5.88647C6.08245 5.83383 3.92583 5.88168 1.77031 5.82906C1.37083 5.78688 0.974633 5.65207 0.587438 5.4266C0.239034 5.25091 0.0260169 4.4051 0.0657282 3.57544C0.0802365 3.27187 0.0835151 2.96627 0.0755221 2.6621C0.0593152 2.37422 0.0360468 2.08902 0.00583733 1.80794C0.211693 1.68853 0.420931 1.60668 0.631687 1.56313C2.21116 1.50998 3.79364 1.36545 5.37098 1.47899C7.02772 1.59957 8.68371 1.52614 10.3396 1.54859C14.1278 1.60068 17.9156 1.66799 21.7032 1.75053C22.6376 1.77378 23.5699 1.94774 24.5023 2.08061C24.7434 2.11977 24.982 2.21437 25.2147 2.36309C25.7285 2.68279 25.9726 3.48442 25.9866 4.74866C25.9835 4.9519 25.99 5.1389 25.9946 5.39444ZM1.01098 5.12095L1.01116 4.96572L0.732258 4.97309C0.732185 5.03473 0.732113 5.09636 0.732038 5.16028L1.01098 5.12095ZM0.98205 2.54457C0.854219 3.37199 0.854223 3.37199 1.06634 3.33442C1.04341 3.07024 1.01467 2.80621 0.986893 2.54444L0.98205 2.54457ZM7.78254 2.02475L7.80301 1.91236L7.31881 1.92515L7.31868 2.03701L7.78254 2.02475ZM7.94252 3.502L7.95696 3.57693L8.21359 3.57015L8.21367 3.49483L7.94252 3.502ZM0.558722 1.85039L0.542114 1.97638L0.905262 1.96678L0.90541 1.84123L0.558722 1.85039Z"
      />
    </svg>
    <svg
      className="c5"
      height="7"
      width="26"
    >
      <path
        d="M25.9946 5.39444C25.6239 6.06393 25.2305 6.19303 24.8209 6.19244C22.1095 6.19103 19.4038 6.23513 16.6954 6.17886C13.8755 6.12554 11.0568 5.95577 8.23701 5.88647C6.08245 5.83383 3.92583 5.88168 1.77031 5.82906C1.37083 5.78688 0.974633 5.65207 0.587438 5.4266C0.239034 5.25091 0.0260169 4.4051 0.0657282 3.57544C0.0802365 3.27187 0.0835151 2.96627 0.0755221 2.6621C0.0593152 2.37422 0.0360468 2.08902 0.00583733 1.80794C0.211693 1.68853 0.420931 1.60668 0.631687 1.56313C2.21116 1.50998 3.79364 1.36545 5.37098 1.47899C7.02772 1.59957 8.68371 1.52614 10.3396 1.54859C14.1278 1.60068 17.9156 1.66799 21.7032 1.75053C22.6376 1.77378 23.5699 1.94774 24.5023 2.08061C24.7434 2.11977 24.982 2.21437 25.2147 2.36309C25.7285 2.68279 25.9726 3.48442 25.9866 4.74866C25.9835 4.9519 25.99 5.1389 25.9946 5.39444ZM1.01098 5.12095L1.01116 4.96572L0.732258 4.97309C0.732185 5.03473 0.732113 5.09636 0.732038 5.16028L1.01098 5.12095ZM0.98205 2.54457C0.854219 3.37199 0.854223 3.37199 1.06634 3.33442C1.04341 3.07024 1.01467 2.80621 0.986893 2.54444L0.98205 2.54457ZM7.78254 2.02475L7.80301 1.91236L7.31881 1.92515L7.31868 2.03701L7.78254 2.02475ZM7.94252 3.502L7.95696 3.57693L8.21359 3.57015L8.21367 3.49483L7.94252 3.502ZM0.558722 1.85039L0.542114 1.97638L0.905262 1.96678L0.90541 1.84123L0.558722 1.85039Z"
      />
    </svg>
    <svg
      className="c5"
      height="7"
      width="26"
    >
      <path
        d="M25.9946 5.39444C25.6239 6.06393 25.2305 6.19303 24.8209 6.19244C22.1095 6.19103 19.4038 6.23513 16.6954 6.17886C13.8755 6.12554 11.0568 5.95577 8.23701 5.88647C6.08245 5.83383 3.92583 5.88168 1.77031 5.82906C1.37083 5.78688 0.974633 5.65207 0.587438 5.4266C0.239034 5.25091 0.0260169 4.4051 0.0657282 3.57544C0.0802365 3.27187 0.0835151 2.96627 0.0755221 2.6621C0.0593152 2.37422 0.0360468 2.08902 0.00583733 1.80794C0.211693 1.68853 0.420931 1.60668 0.631687 1.56313C2.21116 1.50998 3.79364 1.36545 5.37098 1.47899C7.02772 1.59957 8.68371 1.52614 10.3396 1.54859C14.1278 1.60068 17.9156 1.66799 21.7032 1.75053C22.6376 1.77378 23.5699 1.94774 24.5023 2.08061C24.7434 2.11977 24.982 2.21437 25.2147 2.36309C25.7285 2.68279 25.9726 3.48442 25.9866 4.74866C25.9835 4.9519 25.99 5.1389 25.9946 5.39444ZM1.01098 5.12095L1.01116 4.96572L0.732258 4.97309C0.732185 5.03473 0.732113 5.09636 0.732038 5.16028L1.01098 5.12095ZM0.98205 2.54457C0.854219 3.37199 0.854223 3.37199 1.06634 3.33442C1.04341 3.07024 1.01467 2.80621 0.986893 2.54444L0.98205 2.54457ZM7.78254 2.02475L7.80301 1.91236L7.31881 1.92515L7.31868 2.03701L7.78254 2.02475ZM7.94252 3.502L7.95696 3.57693L8.21359 3.57015L8.21367 3.49483L7.94252 3.502ZM0.558722 1.85039L0.542114 1.97638L0.905262 1.96678L0.90541 1.84123L0.558722 1.85039Z"
      />
    </svg>
    <svg
      className="c5"
      height="7"
      width="26"
    >
      <path
        d="M25.9946 5.39444C25.6239 6.06393 25.2305 6.19303 24.8209 6.19244C22.1095 6.19103 19.4038 6.23513 16.6954 6.17886C13.8755 6.12554 11.0568 5.95577 8.23701 5.88647C6.08245 5.83383 3.92583 5.88168 1.77031 5.82906C1.37083 5.78688 0.974633 5.65207 0.587438 5.4266C0.239034 5.25091 0.0260169 4.4051 0.0657282 3.57544C0.0802365 3.27187 0.0835151 2.96627 0.0755221 2.6621C0.0593152 2.37422 0.0360468 2.08902 0.00583733 1.80794C0.211693 1.68853 0.420931 1.60668 0.631687 1.56313C2.21116 1.50998 3.79364 1.36545 5.37098 1.47899C7.02772 1.59957 8.68371 1.52614 10.3396 1.54859C14.1278 1.60068 17.9156 1.66799 21.7032 1.75053C22.6376 1.77378 23.5699 1.94774 24.5023 2.08061C24.7434 2.11977 24.982 2.21437 25.2147 2.36309C25.7285 2.68279 25.9726 3.48442 25.9866 4.74866C25.9835 4.9519 25.99 5.1389 25.9946 5.39444ZM1.01098 5.12095L1.01116 4.96572L0.732258 4.97309C0.732185 5.03473 0.732113 5.09636 0.732038 5.16028L1.01098 5.12095ZM0.98205 2.54457C0.854219 3.37199 0.854223 3.37199 1.06634 3.33442C1.04341 3.07024 1.01467 2.80621 0.986893 2.54444L0.98205 2.54457ZM7.78254 2.02475L7.80301 1.91236L7.31881 1.92515L7.31868 2.03701L7.78254 2.02475ZM7.94252 3.502L7.95696 3.57693L8.21359 3.57015L8.21367 3.49483L7.94252 3.502ZM0.558722 1.85039L0.542114 1.97638L0.905262 1.96678L0.90541 1.84123L0.558722 1.85039Z"
      />
    </svg>
    <svg
      className="c5"
      height="7"
      width="26"
    >
      <path
        d="M25.9946 5.39444C25.6239 6.06393 25.2305 6.19303 24.8209 6.19244C22.1095 6.19103 19.4038 6.23513 16.6954 6.17886C13.8755 6.12554 11.0568 5.95577 8.23701 5.88647C6.08245 5.83383 3.92583 5.88168 1.77031 5.82906C1.37083 5.78688 0.974633 5.65207 0.587438 5.4266C0.239034 5.25091 0.0260169 4.4051 0.0657282 3.57544C0.0802365 3.27187 0.0835151 2.96627 0.0755221 2.6621C0.0593152 2.37422 0.0360468 2.08902 0.00583733 1.80794C0.211693 1.68853 0.420931 1.60668 0.631687 1.56313C2.21116 1.50998 3.79364 1.36545 5.37098 1.47899C7.02772 1.59957 8.68371 1.52614 10.3396 1.54859C14.1278 1.60068 17.9156 1.66799 21.7032 1.75053C22.6376 1.77378 23.5699 1.94774 24.5023 2.08061C24.7434 2.11977 24.982 2.21437 25.2147 2.36309C25.7285 2.68279 25.9726 3.48442 25.9866 4.74866C25.9835 4.9519 25.99 5.1389 25.9946 5.39444ZM1.01098 5.12095L1.01116 4.96572L0.732258 4.97309C0.732185 5.03473 0.732113 5.09636 0.732038 5.16028L1.01098 5.12095ZM0.98205 2.54457C0.854219 3.37199 0.854223 3.37199 1.06634 3.33442C1.04341 3.07024 1.01467 2.80621 0.986893 2.54444L0.98205 2.54457ZM7.78254 2.02475L7.80301 1.91236L7.31881 1.92515L7.31868 2.03701L7.78254 2.02475ZM7.94252 3.502L7.95696 3.57693L8.21359 3.57015L8.21367 3.49483L7.94252 3.502ZM0.558722 1.85039L0.542114 1.97638L0.905262 1.96678L0.90541 1.84123L0.558722 1.85039Z"
      />
    </svg>
    <svg
      className="c6"
      height="7"
      width="26"
    >
      <path
        d="M25.9946 5.39444C25.6239 6.06393 25.2305 6.19303 24.8209 6.19244C22.1095 6.19103 19.4038 6.23513 16.6954 6.17886C13.8755 6.12554 11.0568 5.95577 8.23701 5.88647C6.08245 5.83383 3.92583 5.88168 1.77031 5.82906C1.37083 5.78688 0.974633 5.65207 0.587438 5.4266C0.239034 5.25091 0.0260169 4.4051 0.0657282 3.57544C0.0802365 3.27187 0.0835151 2.96627 0.0755221 2.6621C0.0593152 2.37422 0.0360468 2.08902 0.00583733 1.80794C0.211693 1.68853 0.420931 1.60668 0.631687 1.56313C2.21116 1.50998 3.79364 1.36545 5.37098 1.47899C7.02772 1.59957 8.68371 1.52614 10.3396 1.54859C14.1278 1.60068 17.9156 1.66799 21.7032 1.75053C22.6376 1.77378 23.5699 1.94774 24.5023 2.08061C24.7434 2.11977 24.982 2.21437 25.2147 2.36309C25.7285 2.68279 25.9726 3.48442 25.9866 4.74866C25.9835 4.9519 25.99 5.1389 25.9946 5.39444ZM1.01098 5.12095L1.01116 4.96572L0.732258 4.97309C0.732185 5.03473 0.732113 5.09636 0.732038 5.16028L1.01098 5.12095ZM0.98205 2.54457C0.854219 3.37199 0.854223 3.37199 1.06634 3.33442C1.04341 3.07024 1.01467 2.80621 0.986893 2.54444L0.98205 2.54457ZM7.78254 2.02475L7.80301 1.91236L7.31881 1.92515L7.31868 2.03701L7.78254 2.02475ZM7.94252 3.502L7.95696 3.57693L8.21359 3.57015L8.21367 3.49483L7.94252 3.502ZM0.558722 1.85039L0.542114 1.97638L0.905262 1.96678L0.90541 1.84123L0.558722 1.85039Z"
      />
    </svg>
  </div>
</div>
`;
